### ISSUE TITLE
Method Writer fails to compile proxies for core java interfaces

### ISSUE THREAD

**Issue by alamar** [OpenHFT/Chronicle-Wire#560]
```
    @Test
    public void testCoreClassCanBePassedToMethodReader() throws Exception {
        BinaryWire wire = new BinaryWire(Bytes.allocateElasticOnHeap(128));
        wire.usePadding(true);

        Callable writer = wire.methodWriter(Callable.class);

        writer.call();

        AtomicBoolean b = new AtomicBoolean();

        MethodReader reader = wire.methodReader(new Callable() {
            @Override
            public Object call() {
                b.set(true);
                return null;
            }
        });

        assertFalse(reader instanceof VanillaMethodReader);

        assertTrue(reader.readOne());
        assertTrue(b.get());
    }
```

results in 
```
java.lang.SecurityException: Prohibited package name: java.util.concurrent
	at java.lang.ClassLoader.preDefineClass(ClassLoader.java:655)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:754)
	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
	at java.net.URLClassLoader.defineClass(URLClassLoader.java:473)
	at java.net.URLClassLoader.access$100(URLClassLoader.java:74)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:369)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:363)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(URLClassLoader.java:362)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:418)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:352)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:351)
	at java.lang.Class.forName0(Native Method)
	at java.lang.Class.forName(Class.java:264)
	at net.openhft.chronicle.wire.VanillaMethodWriterBuilder.createInstance(VanillaMethodWriterBuilder.java:211)
	at net.openhft.chronicle.wire.VanillaMethodWriterBuilder.get(VanillaMethodWriterBuilder.java:192)
	at net.openhft.chronicle.wire.VanillaMethodWriterBuilder.build(VanillaMethodWriterBuilder.java:137)
	at net.openhft.chronicle.wire.MarshallableOut.methodWriter(MarshallableOut.java:233)
	at net.openhft.chronicle.wire.MarshallableOut.methodWriter(MarshallableOut.java:213)
	at net.openhft.chronicle.wire.MethodReaderNonStandardInstancesTest.testAnonymousClassCanBePassedToMethodReader(MethodReaderNonStandardInstancesTest.java:36)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:69)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater$1.execute(IdeaTestRunner.java:38)
	at com.intellij.rt.execution.junit.TestsRepeater.repeat(TestsRepeater.java:11)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:35)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:235)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:54)
```

I suggest we prepend "net.openhft.wire.method" to packages starting with java or javax (?), to produce `net.openhft.wire.method.java.lang.concurrent.CallableBinarylightMethodWriter` FQ name.

In the future we would likely have to do that for most interfaces as java package model will likely not let us create classes in other packages'  packages.

Thank @george-ball for this one.

**Comment by alamar** [comment]
It seems that MethodWriter also struggles with parametrized types (such as Consumer) and methods returning values (document is not closed and not read later)

**Comment by hft-team-city** [comment]
Released in [Chronicle-Wire-2.23ea37](https://github.com/OpenHFT/Chronicle-Wire/releases/chronicle-wire-2.23ea37), [BOM-2.23ea119](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.23ea119)

**Comment by hft-team-city** [comment]
Released in [Chronicle-Wire-2.23ea38](https://github.com/OpenHFT/Chronicle-Wire/releases/chronicle-wire-2.23ea38), [BOM-2.23ea121](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.23ea121)
